internal/replication/blockreplicator.go

```diff
@@ -285,6 +311,9 @@ func (br *BlockReplicator) deliverEntries(committedEntries []raftpb.Entry) bool
 				break
 			}
 
+			position = i
```

Comment:
I wondering why you need both `position` and `i`. I assume it is because `break` in line 321. And this lead to next question - why `break` and not `continue`?

Reply:
By the time the for loop ends, `position` will carry the last element in `committedEntries` that delivers a block. Blocks are delivered in `EntryNormal` only at this point. There might be `EntryConfChange` elements in the array past the last block, and `position` avoids them.

The `break` breaks from the switch, which is effectively continues to the next loop iteration.